core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -90,6 +91,46 @@ class LogCleanerIntegrationTest(compressionCodec: String) {
     assertFalse(checkpoints.contains(topics(0)))
   }
 
+  @Test
+  def testCleansCombinedCompactAndDeleteTopic() {
+    val logProps  = new Properties()
+    val retentionMs: Integer = 100000
+    logProps.put(LogConfig.RetentionMsProp, retentionMs: Integer)
+    logProps.put(LogConfig.CleanupPolicyProp, "compact,delete")
+
+    def runCleanerAndCheckCompacted(numKeys: Int): (Log, Seq[(Int, String)]) = {
+      cleaner = makeCleaner(parts = 1, propertyOverrides = logProps, logCleanerBackOffMillis = 1000L)
```

Comment:
For the test to finish faster, perhaps we can reduce logCleanerBackOffMillis to sth like 100ms?